core/image/etc/etc2.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ETC2_H_
+#define ETC2_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum etc_format {
+  // ETC2 Formats
+  ETC2_RGB_U8_NORM,
+  ETC2_RGBA_U8_NORM,
+  ETC2_RGBA_U8U8U8U1_NORM,
+  ETC2_SRGB_U8_NORM,
+  ETC2_SRGBA_U8_NORM,
+  ETC2_SRGBA_U8U8U8U1_NORM,
+
+  // EAC Formats
+  ETC2_R_U11_NORM,
+  ETC2_RG_U11_NORM,
+  ETC2_R_S11_NORM,
+  ETC2_RG_S11_NORM,
+
+  // ETC1 Format
+  ETC1_RGB_U8_NORM,
+};
+
+typedef int64_t etc_error;
```

Comment:
We have to be sure that it's 64 byte

Reply:
I'm not sure why you would need 64 bits?

This code is similar to astc, where we have a simple int: https://github.com/google/agi/blob/99990ab55f883b3a6ea970440dbacbeea634e25d/core/image/astc/astc.h#L24

And  `Etc::Image::EncodingStatus` current max value is `ERROR_ZERO_WIDTH_OR_HEIGHT = 1 << 19`, so I guess this enum fits in an `int` ?

Reply:
True, sorry I think I misread that value somehow. In that case I am converting it to int32. It's better to have an explicit bit number rather than external factors decide it for you.

Reply:
I have actually converted it to uint32_t because after changing the allocation to vector, we don't need allocation failed anymore(as reserve does not return any value). So, we don't have any negative value.